bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: cluster logging operator installation

## Description
[MGDSTRM-3342](https://issues.redhat.com/browse/MGDSTRM-3342)

This PR installs the cluster logging operator addon on data plane clusters. This feature is flag enabled (which defaults to false), as we currently only want to install the operator on production clusters. We also only require application-level logs to be sent to Cloud Watch, and not infra logs. 

**To-Do**
- An accompanying PR is to be created within app interface to enable this feature in production.
- Remove/update any SOPs associated with this process

## Verification Steps
1. Check out this branch and run the service (I ran it against an existing OCM cluster) without specifying the flag to enable the cluster logging operator: `./kas-fleet-manager serve`. Ensure the the operator does not get installed.
2. Run the service using the feature flag, ensuring that the operator does get installed: `./kas-fleet-manager serve  --cluster-logging-operator-addon-id=cluster-logging-operator`
3. Check that the operator has been installed with the correct configuration. You can check this in the OCM UI, or via the OCM CLI. Below is an example out from the CLI:
      ```
      ocm get /api/clusters_mgmt/v1/clusters/<cluster_id>/addons
      
            "parameters": {
              "items": [
                {
                  "id": "use-cloudwatch",
                  "value": "true"
                },
                {
                  "id": "use-app-logs",
                  "value": "true"
                },
                {
                  "id": "use-infra-logs",
                  "value": "false"
                },
                {
                  "id": "use-audit-logs",
                  "value": "false"
                }
              ]
            },
      ```
4. Ensure that the Cluster Logging operator is displaying a status of `Succeded`, which indicated a successful install.
5. Check the `openshift-logging` namespace to ensure that a ClusterLogging and ClusterLogForwarder custom resources have been created: `oc get ClusterLogging -n openshift-logging && oc get ClusterLogForwarder -n openshift-logging`



## Type of change

- [ ] Refactor (A code change that neither fixes a bug nor adds a feature. The work is done to address technical debt) 

## Checklist (Definition of Done)
